spark/src/main/scala/org/apache/spark/sql/delta/TypeWideningMetadata.scala

```diff
@@ -130,11 +139,16 @@ private[delta] object TypeWideningMetadata extends DeltaLogging {
     val changesToRecord = mutable.Buffer.empty[TypeChange]
     val schemaWithMetadata = SchemaMergingUtils.transformColumns(schema, oldSchema) {
       case (_, newField, Some(oldField), _) =>
-        // Record the version the transaction will attempt to use in the type change metadata. If
-        // there's a conflict with another transaction, the version in the metadata will be updated
-        // during conflict resolution. See [[ConflictChecker.updateTypeWideningMetadata()]].
-        val typeChanges =
-          collectTypeChanges(oldField.dataType, newField.dataType, txn.getFirstAttemptVersion)
+        var typeChanges = collectTypeChanges(oldField.dataType, newField.dataType)
+        // The version field isn't used anymore but we need to populate it in case the table doesn't
+        // use the stable feature, as preview clients may then still access the table and rely on
+        // the field being present.
+        if (!txn.protocol.isFeatureSupported(TypeWideningTableFeature)) {
```

Comment:
Why do we check for `TypeWideningTableFeature` not being there instead of `TypeWideningPreviewTableFeature` being there?